src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -663,7 +663,7 @@ private bool TryReadEvent(out NotifyEvent notifyEvent)
                 //         uint32_t len;
                 //         char     name[]; // length determined by len; at least 1 for required null termination
                 //     };
```

Comment:
I believe it's correct as-is.  The position could have been moved past the last valid piece of data, in which case _bufferPos == bufferAvailable.